### PR TITLE
Feature/jobs update

### PR DESCRIPTION
**Pull request by MykhailoKutsyi**
Hello 

added Post Jobs update route
added missed fields to Job - its about 

  @ApiProperty({ example: 5 })
  @IsOptional()
  @IsNumber()
  duration?: number;

  @ApiProperty({ example: 'Month' })
  @IsOptional()
  @IsString()
  durationAmount?: DurationAmount;

added also this properies at db schema
image

updated getAllProperties with durationAmount: string[];
image

### CODE REVIEW

File: src/modules/jobs/dto/update-job.dto.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export default class UpdateJobDto {
+  @ApiProperty({ example: 1 })
+  @IsNumber()
+  id: number;
+
+  @ApiProperty({ example: 'Landing page' })
+  @IsOptional()
+  @IsString()
+  title?: string;
```
Comment (SergKukh) [inline src/modules/jobs/dto/update-job.dto.ts:17]: can we update title of job?

File: src/modules/jobs/job.service.ts [hunk 2]
```diff
@@ -211,4 +212,36 @@ export class JobsService {
       throw new InternalServerErrorException();
     }
   }
+
+  async updateJob(payload: UpdateJobDto, user: UserDto): Promise<void> {
+    try {
+      const { id, skills, ...jobPayload } = payload;
+
```
Comment (SergKukh) [inline src/modules/jobs/job.service.ts]: you need to add validation here. If current user equals job owner

File: src/modules/jobs/dto/update-job.dto.ts (hunk 1 above)
Comment (MykhailoKutsyi) [reply src/modules/jobs/dto/update-job.dto.ts:17]: Yes

**Comment by MykhailoKutsyi** [comment]
Added validation 
image

**Review by SergKukh — APPROVED** [review]

File: src/modules/user/user.controller.ts [hunk 3]
```diff
@@ -149,8 +149,8 @@ export class UserController {
   @Get('/freelancer')
   getFreelancerInformation(
     @Query('search') search: string,
-    @Query('take', new DefaultValuePipe(10), ParseIntPipe) take: number = 10,
-    @Query('page', new DefaultValuePipe(1), ParseIntPipe) page: number = 1,
+    @Query('take', new DefaultValuePipe(10), ParseIntPipe) take = 10,
```
Comment (lyudad) [inline src/modules/user/user.controller.ts]: Снимок экрана 2022-12-07 в 21 38 37

File: src/modules/user/user.controller.ts [hunk 4]
```diff
@@ -149,8 +149,8 @@ export class UserController {
...
+    @Query('page', new DefaultValuePipe(1), ParseIntPipe) page = 1,
```
Comment (lyudad) [inline src/modules/user/user.controller.ts]: Снимок экрана 2022-12-07 в 21 38 37 `DefaultValuePipe(1)` what is 1? why 1? why not 1.78?

**Review by pavlo-chernykh — APPROVED** [review]
approved

File: src/modules/jobs/job.service.ts [hunk 5]
```diff
@@ -211,4 +212,39 @@ export class JobsService {
       throw new InternalServerErrorException();
     }
   }
+
+  async updateJob(payload: UpdateJobDto, user: UserDto): Promise<void> {
+    try {
+      const { id, skills, ...jobPayload } = payload;
+      const job = await this.findOne({ id });
+      if (!(job && job.owner.id === user.id)) {
```
Comment (lyudad) [inline src/modules/jobs/job.service.ts:313]: I see this condition 2d or 3d times. and I don't understand it, if you have a relationship connection how can you have this error? why do you check for specific this error

**Review by lyudad — APPROVED** [review]

**Pull request merged**